### PR TITLE
OCPBUGS-31210: insights client is hard coded to x86_64

### PR DESCRIPTION
**Pull request by chandra0007**
Downloads the correct certificates for etc-pki-entitlement for the target architecture

## Categories

- [x] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `path/to/sample_data.json`

## Documentation

- `path/to/documentation.md`

## Unit Tests

- `path/to/file_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/OCPBUGS-31210 
https://docs.openshift.com/container-platform/4.16/cicd/builds/running-entitled-builds.html
https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

**Comment by prb112** [comment]
/title: OCPBUGS-31210: insights client is hard coded to x86_64

**Comment by tremes** [comment]
Hi @chandra0007 thanks for the PR, but I am not sure this is the best approach (I guess this covers only the arch of the node operator is running on). There's https://issues.redhat.com/browse/CCXDEV-12875 to address this

**Comment by prb112** [comment]
Hey @tremes that's right. only the arch for the node (part of control plane) the operator is running on. We were going to open a feature request for the rest. I'm glad to see the issue, and I've subscribed. If we address this as an interim step, it'd ensure the bug is addressed and fixed in earlier version. Going forward with 4.18+ we'd love to see full multi-arch support. Thanks, Paul

File: pkg/insights/insightsclient/requests.go [hunk 1]
```diff
@@ -192,7 +193,13 @@ func (c *Client) RecvSCACerts(_ context.Context, endpoint string) ([]byte, error
 	if err != nil {
 		return nil, err
 	}
-	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer([]byte(scaArchPayload)))
+	arch := runtime.GOARCH
```
Comment (prb112) [inline pkg/insights/insightsclient/requests.go:196]: Add a log of the arch here.

File: pkg/insights/insightsclient/requests.go [hunk 2]
```diff
@@ -192,7 +193,13 @@ func (c *Client) RecvSCACerts(_ context.Context, endpoint string) ([]byte, error
...
+	if runtime.GOARCH == "amd64" {
+		arch = "x86_64"
+	} else if runtime.GOARCH == "arm64" {
+		arch = "aarch64"
+	}
+	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer([]byte(fmt.Sprintf(scaArchPayload, arch))))
```
Comment (prb112) [inline pkg/insights/insightsclient/requests.go:202]: Before this line add a log of the arch

File: pkg/insights/insightsclient/requests.go [hunk 3]
```diff
@@ -192,7 +193,16 @@ func (c *Client) RecvSCACerts(_ context.Context, endpoint string) ([]byte, error
...
-	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer([]byte(scaArchPayload)))
+	arch := runtime.GOARCH
+	klog.Infof("Arch at run time %s", arch)
+	if runtime.GOARCH == "amd64" {
+		arch = "x86_64"
+		klog.Infof("Arch at if condition %s", arch)
+	} else if runtime.GOARCH == "arm64" {
+		arch = "aarch64"
+		klog.Infof("Arch at elif condition %s", arch)
+	}
```
Comment (prb112) [inline pkg/insights/insightsclient/requests.go:204]: The logs should tell us what we have for an arch value.

**Comment by tremes** [comment]
I would suggest to close this one in favor of https://github.com/openshift/insights-operator/pull/979. We will try to backport it.

**Comment by prb112** [comment]
Thanks @tremes 

I've assigned these to the insights operator component:
https://issues.redhat.com/browse/OCPBUGS-31210 
https://issues.redhat.com/browse/OCPBUGS-32233

We're happy to help

**Pull request closed without merging**